exporters/otlp-http/trace/src/main/java/io/opentelemetry/exporter/otlp/http/trace/OtlpHttpSpanExporter.java

```diff
@@ -112,11 +116,11 @@ public CompletableResultCode export(Collection<SpanData> spans) {
               @Override
               public void onFailure(Call call, IOException e) {
                 spansExportedFailure.add(spans.size());
-                result.fail();
                 logger.log(
                     Level.SEVERE,
                     "Failed to export spans. The request could not be executed. Full error message: "
                         + e.getMessage());
+                result.fail();
```

Comment:
I guess this is to try to solve this flake "org.opentest4j.AssertionFailedError: Contain the string <Failed to export spans. Server responded with HTTP status code 500. Error message: Server error!> ==> None of the 0 captured log events matched the filter predicate" but seems to still be happening. Maybe need to just use awaitility for that assertion due to thread issues.

Reply:
Nice! Thanks for the recommendation. I've been pulling my hair out over that - glad to see there's a pattern for dealing with log threading issues in tests.

Reply:
Didn't fix my problem 😞 

This is pretty mysterious. Have tried a bunch of stuff to try to recreate it locally with no luck.

Reply:
@jack-berg Took a look at the code, while it doesn't explain that I can't repro a failure locally (I'd expect it to be more likely locally thanks to the CPU speed) but could it be because of the ThrottlingLogger? It allows 5 log messages in, so we wouldn't expect a problem but it's definitely a component that could be causing a log message to not get pushed out.

Reply:
FWIW, I feel I've only seen this failure on our Mac OS Java 17 CI test. That doesn't help much, I tried Java 17 locally on a Mac to no problem, but recording for the record.